src/index.js

```diff
@@ -68,9 +67,6 @@ class TCP {
       multiaddrs = [multiaddrs]
     }
     return multiaddrs.filter((ma) => {
-      if (includes(ma.protoNames(), 'ipfs')) {
-        ma = ma.decapsulate('ipfs')
-      }
       return mafmt.TCP.matches(ma)
```

Comment:
Notes from the call today:

It should continue to be valid to pass an ipfs multiaddr, it just needs to check things differently.

- 1. check if it is relay, if it is, skip it
- 2. if not relay, check if it is ipfs, if it is, decapsulate
- 3. check if it is TCP, if it is, include it

Reply:
@diasdavid ii and iii already happen in master, so it should be adding a check for circuit and skipping it if its present (i).